### DISCUSSION TITLE
authenticate from HTTP header

### DISCUSSION THREAD

**Discussion by hmoffatt** [semaphoreui/semaphore#1090] [Q&A] [answered]
Is it possible to have Semaphore take its username from an HTTP header and skip the login step? I have SSO already configured in my web server and so I would like users to be automatically logged in.

**Comment by cf-sewe** [comment]
Yes, that would be great. 
How does your web server forward the user information?

- use of HTTP header
  - what is the name of the header
  - what is the content of the header
  - can the header be verified (JWT signature?)

**Reply by hmoffatt** [reply]
I am using Apache HTTPD with mod_auth_openidc, so it's totally configurable. It could be a simple username passed in an environment variable like X-Remote-User, or the OpenID Connect tokens (JWT) can be passed in as well.

The relevant documentation is here for passing the username:
https://github.com/zmartzone/mod_auth_openidc/blob/master/auth_openidc.conf#L812

Or here for passing the ID token and/or user info:
https://github.com/zmartzone/mod_auth_openidc/blob/master/auth_openidc.conf#L778

I'd be happy enough just to pass through a username in my case.

**Comment by hmoffatt** [comment, marked as answer]
I use the internal OpenID Connect for this now.
